driver-core/src/main/java/com/datastax/driver/core/RegularStatement.java

```diff
@@ -51,21 +51,44 @@ protected RegularStatement() {}
      *
      * @param protocolVersion the protocol version in which the returned values
      * must be serialized for.
+     * @param codecRegistry the {@link CodecRegistry} instance to use.
      * @return the values to use for this statement or {@code null} if there is
      * no such values.
-     *
+     * @throws IllegalArgumentException if one of the values is not of a type
+     * that can be serialized to a CQL3 type
      * @see SimpleStatement#SimpleStatement(String, Object...)
      */
-    public abstract ByteBuffer[] getValues(ProtocolVersion protocolVersion);
+    public abstract ByteBuffer[] getValues(ProtocolVersion protocolVersion, CodecRegistry codecRegistry);
 
     /**
      * Whether or not this statement has values, that is if {@code getValues}
      * will return {@code null} or not.
      *
+     * @param protocolVersion the protocol version in which the returned values
+     * must be serialized for.
+     * @param codecRegistry the {@link CodecRegistry} instance to use.
      * @return {@code false} if {@link #getValues} returns {@code null}, {@code true}
      * otherwise.
      */
-    public abstract boolean hasValues();
+    public abstract boolean hasValues(ProtocolVersion protocolVersion, CodecRegistry codecRegistry);
```

Comment:
Indeed. I added that argument on purpose in anticipation of future use, and to be consistent with other methods in this class. But I'm fine if you prefer to remove it.